pandas/core/aggregation.py

```diff
@@ -415,67 +423,94 @@ def transform(
     ValueError
         If the transform function fails or does not transform.
     """
-    from pandas.core.reshape.concat import concat
-
     is_series = obj.ndim == 1
-
     if obj._get_axis_number(axis) == 1:
         assert not is_series
         return transform(obj.T, func, 0, *args, **kwargs).T
 
-    if isinstance(func, list):
+    if is_list_like(func) and not is_dict_like(func):
+        func = cast(List[AggFuncTypeBase], func)
+        # Convert func equivalent dict
         if is_series:
             func = {com.get_callable_name(v) or v: v for v in func}
         else:
             func = {col: func for col in obj}
 
-    if isinstance(func, dict):
-        if not is_series:
-            cols = sorted(set(func.keys()) - set(obj.columns))
-            if len(cols) > 0:
-                raise SpecificationError(f"Column(s) {cols} do not exist")
-
-        if any(isinstance(v, dict) for v in func.values()):
-            # GH 15931 - deprecation of renaming keys
-            raise SpecificationError("nested renamer is not supported")
-
-        results = {}
-        for name, how in func.items():
-            colg = obj._gotitem(name, ndim=1)
-            try:
-                results[name] = transform(colg, how, 0, *args, **kwargs)
-            except Exception as e:
-                if str(e) == "Function did not transform":
-                    raise e
-
-        # combine results
-        if len(results) == 0:
+    if is_dict_like(func):
```

Comment:
shouldn't this be an elif? otherwise these are going to be converted *again(?

Reply:
In the case func is a list, we replace it with the equivalent dict, so we want to hit both the `if is_list_like` and `if is_dict_like` blocks.